include/EASTL/bitset.h

```diff
@@ -1805,7 +1805,7 @@ EA_RESTORE_GCC_WARNING()
 		: base_type(value)
 	{
 		if((N & kBitsPerWordMask) || (N == 0)) // If there are any high bits to clear... (If we didn't have this check, then the code below would do the wrong thing when N == 32.
-			mWord[kWordCount - 1] &= ~(~static_cast<word_type>(0) << (N & kBitsPerWordMask)); // This clears any high unused bits.
+            mWord[kWordCount - 1] &= ~(static_cast<word_type>(~0) << (N & kBitsPerWordMask)); // This clears any high unused bits.
```

Comment:
As discussed in chat, its preferable to disable the warning here instead of introducing this change.